tests/test_socket_server.py

```diff
@@ -173,13 +136,29 @@ async def check_server_test(self, num_channels: int) -> None:
         data = await self.read()
         self.assertEqual(common.ResponseCode.OK, data[common.Key.RESPONSE])
 
-        reply = await self.read()
-        reply_to_check = reply[common.Key.TELEMETRY]
-        self.check_temperature_reply(reply=reply_to_check, num_channels=num_channels)
+        # Make sure that the mock sensor outputs data for a disconnected
+        # channel.
+        self.srv.command_handler._devices[
+            0
+        ]._disconnected_channel = md_props.disconnected_channel
```

Comment:
Please don't use private variables outside of the class. If you really want access to the devices then the instance variable should be named `devices` and the disconnected channel should be named `disconnected_channel` (or the device should have a public way of setting the private variable, such as a method or property setter). Similarly for the other mock device extra variables (_missed_channels, etc.).

Renaming these to eliminate the leading _ would be a pretty quick global search/replace. My only concern is if any of the mock-device-specific variables could ever collide with the base class. You could eliminate the risk by having a dataclass or other struct to hold the mock-specific configuration (which would be more work). But the ones I looked at don't seem likely to collide anyway.

Reply:
I renamed these, and several other, variables and removed the leading `_` from the names.